lib/IRGen/IRGenMangler.cpp

```diff
@@ -87,19 +87,29 @@ IRGenMangler::mangleTypeForReflection(IRGenModule &IGM,
   llvm::SaveAndRestore<std::function<bool (const DeclContext *)>>
     SymbolicReferencesForLocalTypes(CanSymbolicReference);
   
-  if (IGM.CurSourceFile
-      && !isa<ClangModuleUnit>(IGM.CurSourceFile)
+  if ((!IGM.CurSourceFile || !isa<ClangModuleUnit>(IGM.CurSourceFile))
       && !IGM.getOptions().IntegratedREPL) {
     CanSymbolicReference = [&](const DeclContext *dc) -> bool {
+      // Can only symbolically reference nominal type declarations.
+      auto nominal = dyn_cast<NominalTypeDecl>(dc);
+      if (!nominal || isa<ProtocolDecl>(dc))
```

Comment:
Why can’t we reference protocols?

Reply:
It’s historical and should be easy to fix: protocol descriptors weren’t context descriptors at the time @jckarter added symbolic references, IIRC

Reply:
Also, protocols aren't directly types in the mangling. I'm in fact working on fixing this as part of some other cleanups.